tests/ga/test_extension.py

```diff
@@ -1942,6 +1942,58 @@ def test_correct_exit_code_should_be_set_on_uninstall_cmd_failure(self, *args):
                 self.assertIn("%s=%s" % (ExtCommandEnvVariable.UninstallReturnCode, exit_code), install_kwargs['message'])
                 self.assertIn("%s=%s" % (ExtCommandEnvVariable.UninstallReturnCode, exit_code), enable_kwargs['message'])
 
+    def test_path_should_contain_symlink_path_only_if_ub_20(self, *args):
+
+        test_file_name = "testfile.sh"
+        handler_json = {
+            "installCommand": test_file_name,
+            "uninstallCommand": test_file_name,
+            "updateCommand": test_file_name,
+            "enableCommand": test_file_name,
+            "disableCommand": test_file_name,
+            "rebootAfterInstall": False,
+            "reportHeartbeat": False,
+            "continueOnUpdateFailure": False
+        }
+        manifest = HandlerManifest({'handlerManifest': handler_json})
+
+        # Script prints env variables passed to this process and prints all starting with ConfigSequenceNumber
+        test_file = """
+                printenv | grep PATH
+                """
+
+        base_dir = os.path.join(conf.get_lib_dir(), 'OSTCExtensions.ExampleHandlerLinux-1.0.0', test_file_name)
+        self.create_script(test_file_name, test_file, base_dir)
+
+        test_data = mockwiredata.WireProtocolData(mockwiredata.DATA_FILE_EXT_SINGLE)
+        exthandlers_handler, protocol = self._create_mock(test_data, *args)
+
+        file_that_exists_in_fs = base_dir
+        file_that_does_not_exist_in_fs = "some/random/file"
+
+        for file in [file_that_exists_in_fs, file_that_does_not_exist_in_fs]:
+            with patch("azurelinuxagent.common.osutil.factory.UBUNTU_20_04_IMAGE_PATH", file):
+
+                with patch.object(ExtHandlerInstance, "load_manifest", return_value=manifest):
+                    with patch.object(ExtHandlerInstance, 'report_event') as mock_report_event:
+                        exthandlers_handler.run()
+
+                        for _, kwargs in mock_report_event.call_args_list:
+                            # The output is of the format - 'testfile.sh\n[stdout]PATH=N\n[stderr]'
+                            if test_file_name not in kwargs['message']:
+                                continue
+
+                            python_symlink_path = get_python_symlink_path_if_ubuntu_20_04_image()
+                            if python_symlink_path is not None:
+                                self.assertEqual(file, file_that_exists_in_fs, "File should exist in Filesystem if ubuntu image exists")
+                                self.assertIn("\nPATH={0}".format(conf.get_lib_dir()), kwargs['message'], "Agent lib directory should be prepended to path")
```

Comment:
i think we should assert that the full path of the link is in path, rather than just the agent's dir

Reply:
We dont need to provide the whole link path when we prepend to the path before invoking the command. Rather, we just need to provide the directory of where its stored and the shebang walks the PATH and finds it accordingly. We're writing the symlink in the agent dir so we're just prepending the agent dir to the PATH variable rather than the whole link. 

Eg:

```sh
larohra@larohra-ub16dev:~/test_import/modsDir$ file python
python: symbolic link to /usr/bin/python3

# With just directory prepended to path
larohra@larohra-ub16dev:~/test_import/modsDir$ PATH=`pwd`:$PATH ./mods.py
sys.version_info(major=3, minor=5, micro=2, releaselevel='final', serial=0)
Hello test

# With Dir/python prepended to path
larohra@larohra-ub16dev:~/test_import/modsDir$ PATH=`pwd`/python:$PATH ./mods.py
sys.version_info(major=2, minor=7, micro=12, releaselevel='final', serial=0)
Hello test

```